### PR TITLE
Retreival - stream reset

### PR DESCRIPTION
**Pull request by pradovic**
https://github.com/ethersphere/bee/issues/403

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,4 +1,5 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+cloud.google.com/go v0.31.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (acud) [inline go.sum:2]: revert

File: pkg/retrieval/retrieval.go [hunk 2]
```diff
@@ -183,20 +185,23 @@ func (s *Service) closestPeer(addr swarm.Address, skipPeers []swarm.Address) (sw
 
 func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) error {
 	w, r := protobuf.NewWriterAndReader(stream)
-	defer stream.Close()
+	defer stream.FullClose()
 	var req pb.Request
 	if err := r.ReadMsg(&req); err != nil {
+		_ = stream.Reset()
```
Comment (acud) [inline pkg/retrieval/retrieval.go:191]: this can be also handled inside a defer statement in case you have a named return

Comment (pradovic) [reply pkg/retrieval/retrieval.go]: Done

**Comment by acud** [comment]
.. deepsource

**Comment by pradovic** [comment]
> .. deepsource

It's evil! :)

**Comment by acud** [comment]
:D

**Pull request merged**